plugin/src/main/java/org/screamingsandals/bedwars/game/GameManagerImpl.java

```diff
@@ -87,17 +88,32 @@ public boolean hasGame(UUID uuid) {
     }
 
     @Override
-    public Optional<GameImpl> getGameWithHighestPlayers(boolean fee) {
-        return games.stream()
-                .filter(game -> game.getStatus() == GameStatus.WAITING)
-                .filter(game -> game.countConnectedPlayers() < game.getMaxPlayers())
-                .filter(game -> {
-                    if (fee) {
-                        return game.getFee() > 0;
+    public Optional<GameImpl> getGameWithHighestPlayers(boolean fee) {  // If tie choose random one
+        List<GameImpl> highestCountGames = new ArrayList<>();
+        GameManagerImpl.getInstance().getGames().stream()
+                .filter(waitingGame -> waitingGame.getStatus() == GameStatus.WAITING)
+                .filter(waitingGame -> waitingGame.getFee() > 0 || !fee)
+                .forEach(waitingGame -> {
+                    if (highestCountGames.isEmpty()) {
+                        highestCountGames.add(waitingGame);
                     }
-                    return true;
-                })
-                .max(Comparator.comparingInt(GameImpl::countConnectedPlayers));
+                    var playerCount = waitingGame.countPlayers();
```

Comment:
this could be cleaned up a little bit with a custom Collector, like in this [stackoverflow answer](https://stackoverflow.com/a/29339106/19208782)

Reply:
Does [this commit](https://github.com/ScreamingSandals/BedWars/pull/421/commits/21dc293c501c05c5c56d0a9cd1260bfa8259b903) implement it the way you're talking about?

Reply:
not exactly how I wanted it, but looks much better 👍